hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpcengine/nio/NioAsyncSocketOptions.java

```diff
@@ -115,43 +116,76 @@ public <T> void set(Option<T> option, T value) {
-                if (JDK_NET_TCP_KEEPIDLE == null) {
-                    if (TCP_KEEPIDLE_PRINTED.compareAndSet(false, true)) {
-                        logger.warning("Ignoring TCP_KEEPIDLE. "
-                                + "Please upgrade to Java 11+ or configure tcp_keepalive_time in the kernel. "
-                                + "For more info see https://tldp.org/HOWTO/html_single/TCP-Keepalive-HOWTO/. "
-                                + "If this isn't dealt with, idle connections could be closed prematurely.");
-                    }
-                } else {
-                    channel.setOption(JDK_NET_TCP_KEEPIDLE, (Integer) value);
-                }
+                setTcpKeepIdle((Integer) value);
             } else if (TCP_KEEPINTERVAL.equals(option)) {
-                if (JDK_NET_TCP_KEEPINTERVAL == null) {
-                    if (TCP_KEEPINTERVAL_PRINTED.compareAndSet(false, true)) {
-                        logger.warning("Ignoring TCP_KEEPINTERVAL. "
-                                + "Please upgrade to Java 11+ or configure tcp_keepalive_intvl in the kernel. "
-                                + "For more info see https://tldp.org/HOWTO/html_single/TCP-Keepalive-HOWTO/. "
-                                + "If this isn't dealt with, idle connections could be closed prematurely.");
-                    }
-                } else {
-                    channel.setOption(JDK_NET_TCP_KEEPINTERVAL, (Integer) value);
-                }
+                setTcpKeepInterval((Integer) value);
             } else {
                 throw new UnsupportedOperationException("Unrecognized option:" + option);
             }
         } catch (IOException e) {
             throw new UncheckedIOException("Failed to setOption [" + option.name() + "] with value [" + value + "]", e);
         }
     }
+
+    private <T> void setTcpKeepInterval(Integer value) throws IOException {
+        if (JDK_NET_TCP_KEEPINTERVAL == null) {
+            if (TCP_KEEPINTERVAL_PRINTED.compareAndSet(false, true)) {
+                if (OS.isLinux()) {
+                    logger.warning("Ignoring TCP_KEEPINTERVAL. "
+                            + "Please upgrade to Java 11+ and Linux or configure tcp_keepalive_intvl in the kernel. "
+                            + "For more info see https://tldp.org/HOWTO/html_single/TCP-Keepalive-HOWTO/. "
+                            + "If this isn't dealt with, idle connections could be closed prematurely.");
+                } else if (OS.isWindows()) {
+                    logger.warning("Ignoring TCP_KEEPINTERVAL. This option is not supported on Windows.");
+                } else {
+                    logger.warning("Ignoring TCP_KEEPINTERVAL. The option either isn't supported by the OS "
+                            + "or by the JVM (Java 11+ required).");
+                }
+            }
+        } else {
+            channel.setOption(JDK_NET_TCP_KEEPINTERVAL, value);
+        }
+    }
```

Comment:
```suggestion
private <T> void setTcpKeepInterval(Integer value) throws IOException {
         if (OS.isWindows()) {
             if (TCP_KEEPINTERVAL_PRINTED.compareAndSet(false, true)) {
                 logger.warning("Ignoring TCP_KEEPINTERVAL. This option is not supported on Windows.");
             }
             return;
        }
    
        if (JDK_NET_TCP_KEEPINTERVAL == null) {
            if (TCP_KEEPINTERVAL_PRINTED.compareAndSet(false, true)) {
                if (OS.isLinux()) {
                    logger.warning("Ignoring TCP_KEEPINTERVAL. "
                            + "Please upgrade to Java 11+ and Linux or configure tcp_keepalive_intvl in the kernel. "
                            + "For more info see https://tldp.org/HOWTO/html_single/TCP-Keepalive-HOWTO/. "
                            + "If this isn't dealt with, idle connections could be closed prematurely.");
                } else {
                    logger.warning("Ignoring TCP_KEEPINTERVAL. The option either isn't supported by the OS "
                            + "or by the JVM (Java 11+ required).");
                }
            }
        } else {
            channel.setOption(JDK_NET_TCP_KEEPINTERVAL, value);
        }
    }
```
I think it should be something like this. If not we would need to test what happens on windows when `JDK_NET_TCP_KEEPINTERVAL` isn't null.

Reply:
You are totally right. Ok, lets play the build bingo again.